examples/custom-widget-with-3rd-party-data/src/custom-widget/custom-widget.tsx

```diff
@@ -0,0 +1,66 @@
+import { AuthState, DDClient } from '@datadog/ui-extensions-sdk';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { User } from '../user';
+import { Post } from '../post';
+
+type CustomWidgetProps = {
+    client: DDClient<User>;
+};
+
+/**
+ * This component brings together all the pieces and renders a custom widget.
+ */
+function CustomWidget(props: CustomWidgetProps): JSX.Element {
+    /**
+     * We grab the {@link User} from the {@link AuthState}.
+     */
+    const [user, setUser] = React.useState<User>();
+    React.useEffect(() => {
+        props.client.auth
+            .getAuthState()
+            .then((authState: AuthState<User>): void => {
+                setUser(authState.args);
+            });
+    }, [props.client.auth]);
+
+    /**
+     * If we do not have a {@link User},
+     * then authentication was not successful.
+     * Datadog will not actually render the custom widget in this case,
+     * but we still have to handle this case.
+     */
+    if (user == null) {
+        return <></>;
+    }
```

Comment:
This is interesting. I feel like the whole point of our auth system is that the app already has information about the user, and just wants to let us know about it so we can display a connect button or not. Is it the right pattern to get that data from the auth state args instead of directly from localstorage?

I have to be honest that the utility of auth state args and also `getAuthState()` has always felt murky to me, we sorta did it from a 'why not?' point of view and to be consistent with the args pattern for sidepanels + modals

Reply:
This is good background! As someone that wasn't around for these discussions, I assumed we wanted App developers to store information in `AuthState` so they can use it in a different place in the App. Interesting to know that was not a hard requirement. I don't personally have a preference one way or the other. I think some things are harder one way and easier the other way.

However, I think if we don't expect or want App developers to store things in `AuthState`, we should drop it in exchange for something else (likely a `boolean`), get rid of the rest of the stuff like `DDAuthClient.getAuthState`/`DDAuthClient.updateAuthState`, and reword the methods to focus on auth rather than the state of the auth (e.g. rename `authStateCallback` to `isAuthenticated`). If all the App is able to do is say whether the user is authenticated or not, it seems hard to justify methods for other things around the auth state, or even hint at the idea that an App might be visible if it requires 3rd-party auth, but isn't actually authenticated.

For the sake of moving this template forward, I'm going to change this stuff to not use the `AuthState`.

Reply:
I'm open to removing those hooks + auth args, I'll make a ticket